src/Uno.UI.Runtime.Skia.Gtk/Extensions/GtkDisplayInformationExtension.cs

```
@@ -10,27 +10,32 @@ namespace Uno.UI.Runtime.Skia;
 internal class GtkDisplayInformationExtension : IDisplayInformationExtension
 {
 	private readonly DisplayInformation _displayInformation;
-	private readonly Window _window;
 	private readonly DpiHelper _dpiHelper;
 
 	private float? _dpi;
 
-	public GtkDisplayInformationExtension(object owner, Gtk.Window window)
+	public GtkDisplayInformationExtension(object owner)
 	{
 		_displayInformation = (DisplayInformation)owner;
-		_window = window;
-		_dpiHelper = new DpiHelper(_window);
+		GtkHost.Current!.MainWindowShown += GtkDisplayInformationExtension_MainWindowShown;
+		_dpiHelper = new DpiHelper();
+	}
+
+	private void GtkDisplayInformationExtension_MainWindowShown(object? sender, EventArgs e)
+	{
 		_dpiHelper.DpiChanged += OnDpiChanged;
 	}
 
+	private Gtk.Window GetWindow() => GtkHost.Current!.MainWindow!;
```

Let's fail properly if the window is not available.